src/utils.rs

```diff
@@ -597,7 +599,9 @@ pub(crate) fn trim_left_preserve_layout(
                 trimmed = false;
                 line
             } else {
-                line.trim().to_owned()
+                line.trim_start_matches(|c: char| c.is_whitespace() && c != NON_BREAKING_SPACE)
```

Comment:
Can we use is_ascii_whitespace()? It will give false for non_breaking_space and true for space https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=e29f4e7e21de978a039dca17e7510ff2

Reply:
Thanks for the review! I think that's a great idea, and would be much simpler. I'll update this to use `is_ascii_whitespace`